Fatras/include/ActsFatras/Digitization/UncorrelatedHitSmearer.hpp

```diff
@@ -62,12 +64,18 @@ struct BoundParametersSmearer {
   Result operator()(generator_t& rng, const Hit& hit,
                     const Acts::Surface& surface,
                     const Acts::GeometryContext& geoCtx) const {
+    // We use the thickness of the detector element as tolerance, because Geant4
+    // treats the Surfaces as volumes and thus it is not ensured, that each hit
+    // lies exactely on the Acts::Surface
+    assert(surface.associatedDetectorElement());
+    const auto tolerance = surface.associatedDetectorElement()->thickness();
+
     // construct full bound parameters. they are probably not all needed, but it
     // is easier to just create them all and then select the requested ones.
     Acts::Result<Acts::BoundVector> boundParamsRes =
-        Acts::detail::transformFreeToBoundParameters(hit.position(), hit.time(),
-                                                     hit.unitDirection(), 0,
-                                                     surface, geoCtx);
+        Acts::detail::transformFreeToBoundParameters(
+            hit.position(), hit.time(), hit.unitDirection(), 0, surface, geoCtx,
+            tolerance);
```

Comment:
does it actually make sense to use `transformFreeToBoundParameters` here? the surface method might be sufficient no?

(sorry this is of course what was there before)

Reply:
can we leave it as it is?